src/supernova/mod.rs

```diff
@@ -1045,6 +1047,20 @@ pub fn circuit_digest<
   circuit_params.digest()
 }
 
+/// Compute the number of absorbs for the random-oracle computing the circuit output
+/// X = H(vk, i, pc, z0, zi, U)
+fn num_ro_inputs(num_circuits: usize, num_limbs: usize, arity: usize, is_primary: bool) -> usize {
```

Comment:
Finally, thank you. This could also reasonably be a method of `SuperNovaAugmentedCircuit` with even fewer arguments.

Reply:
I wanted to do that originally, but the function is used by `RecursiveSNARK::verify`, where we don't create a `SuperNovaAugmentedCircuit` instance.